src/__tests__/errors.test.ts

```diff
@@ -0,0 +1,42 @@
+import {instance as globalContainer} from "../dependency-container";
+import {inject, injectable} from "../decorators";
+
+afterEach(() => {
+  globalContainer.reset();
+});
+
+test("Error message composition", () => {
+  class Ok {}
+
+  @injectable()
+  class C {
+    constructor(public s: any) {}
+  }
+
+  @injectable()
+  class B {
+    constructor(public c: C) {}
+  }
+
+  @injectable()
+  class A {
+    constructor(public d: Ok, public b: B) {}
+  }
+  expect(() => {
+    globalContainer.resolve(A);
+  }).toThrow(
+    /Cannot inject the dependency "b" at position #1 of "A" constructor. Reason:\s+Cannot inject the dependency "c" at position #0 of "B" constructor. Reason:\s+Cannot inject the dependency "s" at position #0 of "C" constructor. Reason:\s+TypeInfo not known for "Object"/
```

Comment:
nit: Any way we can format this to be a little more readable?

Reply:
`[/ /,/ /].map(x=>x.source).join()`  as in  https://github.com/microsoft/tsyringe/pull/88/commits/b11ea0cb9c51adced5fb8a253d20a299fe1ef860

 makes readability any better?